### PR TITLE
When I added a time entry by mistake, I want to be able to delete it #18

### PR DESCRIPTION
**Pull request by seantijmons**
- Update the list of time entries
- delete api

Both are already working with redux in next branch

### CODE REVIEW

File: components/time-entries/TimeEntries.jsx [hunk 1]
```diff
@@ -30,6 +30,22 @@ const TimeEntries = ({ timeEntries }) => (
           <div className="time-entry__client">
             {entry.client}
           </div>
+          <button
+            className="time-entry__deletebutton"
```
Comment (boris-arkenaar) [inline components/time-entries/TimeEntries.jsx]: `delete-button` (two words).

File: components/time-entries/TimeEntries.jsx [hunk 2]
```diff
@@ -30,6 +30,22 @@ const TimeEntries = ({ timeEntries }) => (
...
+            type="button"
+            onClick={() => {
+              if (window.confirm('Are you sure you wish to delete this item?')) deleteTimeEntry(entry.id);
```
Comment (boris-arkenaar) [inline components/time-entries/TimeEntries.jsx]: You could write `window.confirm('Are you sure you wish to delete this item?') && deleteTimeEntry(entry.id)`

File: components/time-entries/TimeEntries.jsx [hunk 3]
```diff
@@ -30,6 +30,22 @@ const TimeEntries = ({ timeEntries }) => (
...
+            }}
+          >
+            <object
```
Comment (boris-arkenaar) [inline components/time-entries/TimeEntries.jsx]: Why an object? Why not define the icon as a background in your style sheet?

File: components/time-sheet/TimeSheet.jsx [hunk 4]
```diff
@@ -34,11 +34,18 @@ class TimeSheet extends React.Component {
...
+  }
+
   render() {
     return (
       <div className="time-entries">
         <NewTimeEntry addTimeEntry={this.addTimeEntry} />
-        <TimeEntries timeEntries={this.state.timeEntries} />
+        <TimeEntries
+          timeEntries={this.state.timeEntries}
+          addTimeEntry={this.addTimeEntry}
```
Comment (boris-arkenaar) [inline components/time-sheet/TimeSheet.jsx]: `onAddTimeEntry={this.handleAddTimeEntry}`

File: components/time-entries/time-entries.scss [hunk 5]
```diff
@@ -30,9 +30,44 @@
     }
   }
 
+  &:hover {
+    .time-entry {
```
Comment (boris-arkenaar) [inline components/time-entries/time-entries.scss]: `.time-entry` inside `.time-entry`?

File: components/time-sheet/TimeSheet.jsx (hunk 4 above)
Comment (seantijmons) [reply components/time-sheet/TimeSheet.jsx]: already fixed on next branch

File: components/time-entries/TimeEntries.jsx [hunk 6]
```diff
@@ -31,10 +31,11 @@ const TimeEntries = ({ timeEntries, deleteTimeEntry }) => (
             {entry.client}
           </div>
           <button
-            className="time-entry__deletebutton"
+            className="time-entry__delete-button"
             type="button"
             onClick={() => {
-              if (window.confirm('Are you sure you wish to delete this item?')) deleteTimeEntry(entry.id);
+              if (window.confirm('Are you sure you wish to delete this item?')
+              && deleteTimeEntry(entry.id));
```
Comment (boris-arkenaar) [inline components/time-entries/TimeEntries.jsx]: This is an `if` statement with only a condition, without a code block. That's strange.
What I meant was, make it a single expression, without using an `if` statement.
```
onClick={() => (window.confirm('Are you sure you wish to delete this item?')
              && deleteTimeEntry(entry.id))}
```

**Pull request merged**